CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+Package Contribution Guidelines
+===============================
+
+If you are submitting a new Spack package, please observe the
+following standards:
+
+1. Code must be [PEP-8 compliant](http://spack.readthedocs.io/en/latest/contribution_guide.html?highlight=pep-8#flake8-tests)
+
+2. At least one numeric (checksummable) version must be supplied.  If
+   this package does not yet have any official releases, create an
+   unofficial checksummable Spack-only release using something like:
+   ```
+   version('1.9.5.1.1',
+        git='https://github.com/jswhit/pyproj.git',
+        commit='0be612cc9f972e38b50a90c946a9b353e2ab140f')
+   ```
+
+2. All explicit versions in `depends_on()` statements must be for
+   checksummable versions.  If you need to depend on unreleased
+   versions of other packages, add unofficial releases to them too.
+
+3. If the package builds binaries... once you've built the package on
```

Comment:
I think it's hard to automate, for a few reasons:

1. You have to actually build the package.
2. You have to have a good sense of what the binaries are.  (Yes, you could scan for ELF files, but that might not always do what we want).
3. You need to use discretion interpreting the result.  There will always be lots of system-supplied libraries in any real-life build.